tests/utils.py

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
+
+import torch.nn as nn
+from poptorch.enums import CommGroupType, VariableRetrievalMode
+
+
+def _apply_replica_grouping(
```

Comment:
If merging the tests, ignore this, but otherwise, it seems odd to have only _private functions in a module—perhaps just export this.